runtime/lua/vim/lsp.lua

```diff
@@ -1573,7 +1591,23 @@ function lsp.start_client(config)
     if method ~= 'textDocument/didChange' then
       changetracking.flush(client)
     end
-    return rpc.notify(method, params)
+
+    local result = rpc.notify(method, params)
+
+    vim.schedule(function()
+      nvim_exec_autocmds('LspNotify', {
+        modeline = false,
+        data = {
+          request = {
+            method = method,
+            params = params,
+            result = result,
+          },
+        },
```

Comment:
I'm aware this is modelled after `LspRequest` but I'm not sure if there's much value in the extra `request` nesting.

What do you think of having `method`, `params` and `result` directly under `data` ?

Actually, I'd probably remove `result` and instead only invoke the autocmd if it's true (false indicates the client stopped, so in a way there was no notification)
`result` is also a bit missleading, as notifications themselves have no result.

This will also need docs similar to how `LspRequest` is documented

Reply:
Looks good to me. Would passing along the full client also work?

```
data = {
  client = client,
  method = method,
  params = params,
}
```

If it's limited to primitive value it's also fine to go with the client_id. In case the client works it's probably more convenient to have access to it.

Reply:
I think passing `client` does work. But `LspAttach` currently passes `client_id`. There was a PR to change that, but the author vanished.

I'm in favor of passing the full `client`. Just keep in mind that will introduce inconsistency with `LspAttach` (but that inconsistency can, and should, be fixed).

Reply:
Trying to pass `client` is giving me errors:
```lua
Error executing vim.schedule lua callback: /home/catlee/src/neovim/runtime/lua/vim/lsp.lua:1599: Cannot convert given lua table
stack traceback:
        [C]: in function 'nvim_exec_autocmds'
        /home/catlee/src/neovim/runtime/lua/vim/lsp.lua:1599: in function </home/catlee/src/neovim/runtime/lua/vim/lsp.lua:1598>
```

Reply:
Hmm, ok, maybe there _were_ reasons we couldn't pass the full client. I remember it had something to do with not being able to serialize the client table since it contains a `userdata` type. But for some reason I thought that had either been resolved or was a non-issue (apparently that is not the case).